src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs

```diff
@@ -17,46 +17,65 @@ namespace Microsoft.Health.Fhir.TemplateManagement.Client
     public class OrasClient : IOrasClient
     {
         private readonly string _imageReference;
+        private readonly Regex _digestRegex = new Regex("(?<algorithm>[A-Za-z][A-Za-z0-9]*([+.-_][A-Za-z][A-Za-z0-9]*)*):(?<hex>[0-9a-fA-F]{32,})");
 
         public OrasClient(string imageReference)
         {
             EnsureArg.IsNotNull(imageReference, nameof(imageReference));
 
             _imageReference = imageReference;
+            if (string.IsNullOrEmpty(Environment.GetEnvironmentVariable(Constants.OrasCacheEnvironmentVariableName)))
+            {
+                Environment.SetEnvironmentVariable("ORAS_CACHE", Constants.DefaultOrasCacheEnvironmentVariable);
+            }
         }
 
         public async Task PullImageAsync(string outputFolder)
         {
             string command = $"pull  \"{_imageReference}\" -o \"{outputFolder}\"";
-            await OrasExecutionAsync(command, Directory.GetCurrentDirectory());
+            string output = await OrasExecutionAsync(command, Directory.GetCurrentDirectory());
+            Console.WriteLine(output);
+            string digest;
+            try
+            {
+                digest = _digestRegex.Matches(output)[0].Groups["hex"].ToString();
+            }
+            catch (Exception ex)
+            {
+                throw new OrasException(TemplateManagementErrorCode.OrasProcessFailed, "Return image digest failed.", ex);
+            }
+
+            File.WriteAllText(Path.Combine(outputFolder, Constants.ManifestFileName), LoadManifestContentFromCache(digest));
```

Comment:
Does output folder always exist?

Reply:
If Oras pull image successfully, the output folder will exist. Otherwise, it will throw exception before this operation. 
But here, I also add the double check.